pkg/util/topsql/reporter/pubsub.go

```diff
@@ -129,6 +131,8 @@ func (ds *pubSubDataSink) run() error {
 			// In order to clean up resources as quickly as possible, we let that closure run in an individual goroutine,
 			// and wait for timeout here.
 			<-ctx.Done()
+			// useless, it is called to prevent linter error
```

Comment:
How about change `ctx, rcancel := context.WithDeadline(ds.ctx, task.deadline)` to `ctx, _ := context.WithDeadline(ds.ctx, task.deadline)`? The lint will report error or not?

Reply:
Yes, and I don't want it ignore this file completely. Or do you think that is better?

Reply:
Maybe use `defer rcanel()` in Line 113 instead.

Reply:
It is not in a function, but in a loop. I think it will spawn infinite number of deferred goroutines as long as there is no failure.